src/index.ts

```diff
@@ -60,9 +60,8 @@ export class PromiseQueue {
 				}
 				let [timeoutSeconds] = process.hrtime();
 				timeoutSeconds += maxAgeSeconds;
-
 				const firstValid = this.queue.findIndex(
-					({ arrivalTime }) => arrivalTime[0] > timeoutSeconds,
+					({ arrivalTime }) => arrivalTime > timeoutSeconds,
```

Comment:
`timeoutSeconds` is in seconds but `arrivalTime` is in nanoseconds right?  So this condition will always be true and never time anything out?

Reply:
Is it alright to divide inside this anonymous function, given that it runs for each `findIndex()` iteration? or should we just create a BigInt `timeoutNano = timeoutSeconds * 1e9`?

Reply:
(I prefer the latter)

Reply:
We should avoid anything other than the bare minimum inside this filter since it has the ability to run multiple times, ie running it outside is `O(1)` and inside is `O(n)`

Reply:
Yep, makes sense. Push incoming